test/integration/integration_test.go

```diff
@@ -7,11 +7,12 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"os/exec"
```

Comment:
I see that this integration test runs `kubectl` CLI to interact with the cluster, is there any reason to go with this approach instead of using the [go kubernetes client ](https://github.com/kubernetes/client-go) that we are already using in the operator, the apiserver and the sink?

Reply:
I use the go client to check the status of the core objects (Deployment, Replica set and the Pod). But for the ApiServerSource and the EvenType, I tried to set up a [Dynamic Client](https://pkg.go.dev/k8s.io/client-go/dynamic) to interact with those objects but I couldn't manage to set that correctly. So, I decided to check the status of those using kubectl.

Reply:
I see that we are also using `kubectl`  and `ko` for creating, deleting and retrieving resources from the cluster in the [main file of the test package](https://github.com/kubearchive/kubearchive/blob/main/test/main.go#L67) so I guess it's ok to use it here too.

If we want to add a refactor for that in the future and use a go client for everything we can address that in a separate issue.

I've created https://github.com/kubearchive/kubearchive/issues/275 for that.

Reply:
Just to add a little bit of context...

On of the reasons to use the kubectl command is to deal with non-standard CRDs, its a pain to build the objects using the unstructured structs. On the other hand, building the YAML file and apply it is easier. We can use the go client for everything, but sometimes that is a pain.

And ko is more of a residue of the first implementation, that applied one of our images using `ko` to build it before using it. That can be removed for sure.